thirdparty_builtin/CMakeLists.txt

```diff
@@ -181,3 +153,32 @@ if(ENABLE_TESTS)
     endif()
 
 endif()
+
+if(ENABLE_BENCHMARKS)
+    if(WIN32 AND BUILD_SHARED_LIBS)
+      message(FATAL_ERROR "Benchmarks cannot be built when BUILD_SHARED_LIBS=On")
+    endif()
+
+    ## google benchmark support
+    add_subdirectory(gbenchmark-master-2017-05-19
+                     ${BLT_BUILD_DIR}/thirdparty_builtin/gbenchmark-master-2017-05-19)
+
+    if (UNIX AND NOT APPLE)
+      find_library(RT_LIBRARIES rt)
+    endif()
+
+    blt_register_library(NAME gbenchmark
+                         INCLUDES ${benchmark_SOURCE_DIR}/include ${benchmark_SOURCE_DIR}
+                         LIBRARIES benchmark ${RT_LIBRARIES}
+                         TREAT_INCLUDES_AS_SYSTEM ON
+                         )
+
+    list(APPEND _blt_tpl_targets benchmark)
+
+    # This sets up a target to run the benchmarks
+    add_custom_target(run_benchmarks 
+                      COMMAND ctest -C Benchmark -VV -R benchmark
+                      WORKING_DIRECTORY ${PROJECT_BINARY_DIR}
+                      )
+endif()
```

Comment:
Please add the ``if(ENABLE_FOLDERS)`` part for benchmarks.

Reply:
To clarify:  The ``if(ENABLE_FOLDERS)`` part has nothing to do with testing or benchmarks.  It is a feature for multi-config generators in IDE's like visual studio that helps organize the code into hierarchical folders when you have a lot of projects.